### PR TITLE
Hyperopt: Add export CSV-file option

### PR DESCRIPTION
**Pull request by Fredrik81**
## Summary
Add options for exporting CSV-file from hyperopt

Work in progress.
Tests are missing at the moment.

## Quick changelog

- New command to hyperopt-list: --export-csv FILE

## What's new?
freqtrade hyperopt-list --export-csv /tmp/test.csv
or
<s>freqtrade hyperopt-list --export-csv +/tmp/test.csv</s> (this functionality refused; edited by @hroff-1902)

### CODE REVIEW

File: freqtrade/optimize/hyperopt.py [hunk 1]
```diff
@@ -381,6 +383,62 @@ def print_result_table(config: dict, results: list, total_epochs: int, highlight
...
+    @staticmethod
+    def export_csv_file(config: dict, results: list, total_epochs: int, highlight_best: bool,
+                        csv_file: str, overwrite: bool) -> None:
+        """
+        Log result to csv-file
+        """
+        if not results:
+            return
+
+        # Verification for owerwrite
```
Comment (hroff-1902) [inline freqtrade/optimize/hyperopt.py]:
```suggestion
        # Verification for overwrite
```

Comment (Fredrik81) [reply freqtrade/optimize/hyperopt.py]: Fixed in other commit.. thanks

**Comment by hroff-1902** [comment]
@xmatthias I would like to see your comment here about the "+/tmp/test.csv" syntax (the '+' sign).
We do not use it with other options accepting file names, so if this will be implemented here, we'll than need to add same to other options, for better user experience (same syntax and same behavior)...

So I have doubts about this...

**Comment by hroff-1902** [comment]
^^ the `+/tmp/test.csv` syntax discussed at Slack (core-dev channel), refused

**Comment by hroff-1902** [comment]
https://github.com/freqtrade/freqtrade/pull/3021/commits/90ba9055e181d65a860e4b476ce4aa84478e1ec8 is wrong (compare with output of current develop)

File: freqtrade/optimize/hyperopt.py [hunk 2]
```diff
@@ -381,6 +383,62 @@ def print_result_table(config: dict, results: list, total_epochs: int, highlight
...
+                        csv_file: str) -> None:
+        """
+        Log result to csv-file
+        """
+        if not results:
+            return
+
+        # Verification for overwrite
+        if path.isfile(csv_file):
+            logging.error("CSV-File already exists!")
```
Comment (hroff-1902) [inline freqtrade/optimize/hyperopt.py]:
```suggestion
            logger.error("CSV-File already exists!")
```
'logging' does not print the module issuing the message properly...
same below (check for other appearances)

Comment (Fredrik81) [reply freqtrade/optimize/hyperopt.py]: Hmm yes that was a miss.. strange that it worked :-)

File: freqtrade/optimize/hyperopt.py [hunk 3]
```diff
@@ -381,6 +383,62 @@ def print_result_table(config: dict, results: list, total_epochs: int, highlight
...
+
+        trials['Total profit'] = trials['Total profit'].apply(
+            lambda x: '{:,.8f}'.format(x) if x != 0.0 else ""
+        )
+        trials['Profit'] = trials['Profit'].apply(
+            lambda x: '{:,.2f}'.format(x) if not isna(x) else ""
+        )
+        trials['Avg profit'] = trials['Avg profit'].apply(
+            lambda x: ('{:,.2f}%'.format(x)) if not isna(x) else ""
+        )
```
Comment (xmatthias) [inline freqtrade/optimize/hyperopt.py]: the 2 lambda functions seem to do the same thing - but parentheses are different ... ?

also, maybe using an f-string to be aligned to most of the other strings we have would make sense (?) but that's purely optional

**Pull request merged**